### PR TITLE
fix(model_schema): Made changes to model schema

### PR DESCRIPTION
**Pull request by tanushree04**
Updated internal loads in RoomEnergyProperties.
Updated IdealAirSystem Schema heating_limit, cooling_limit

### CODE REVIEW

File: app/models/energy/idealair.py
Comment (chriswmackey) [inline app/models/energy/idealair.py]: This should be:
```
if v != 'autosize' or None:
```
We interpret the `None` input as corresponding to no limit.

File: app/models/energy/idealair.py
Comment (chriswmackey) [inline app/models/energy/idealair.py]: same here.

**Review by chriswmackey — APPROVED** [review]
Thanks for making these fixes @tanushree04 .  The two changes that requested on the `IdealAirSystem` are important but, once you fix those, you can merge this in.

**Comment by chriswmackey** [comment]
Ant it looks like your tests are still failing because you use the word `programset` somewhere:
https://travis-ci.org/ladybug-tools-in2/energy-model-schema/builds/609446720?utm_source=github_status&utm_medium=notification

File: app/models/energy/programtype.py
Comment (tanushree04) [inline app/models/energy/programtype.py:30]: @chriswmackey Does the name field validation look okay?

**Comment by tanushree04** [comment]
@chriswmackey I added the validation for `name` to some energy objects. I can add it to the remaning energy objects once you review it.
You can go ahead and review the `SimulationParameters` schema. The example files you added are passing validation against the schema.

File: app/models/energy/programtype.py
Comment (chriswmackey) [reply app/models/energy/programtype.py:30]: Thanks for doing this @tanushree04 . However, it looks like your code here is trying to reformat the name into a valid one rather than just checking whether the name is valid to begin with. I should have clarified that the original Python code that I sent is intended to format a "display name" input by a user into a valid name that can be sent to EnergyPlus.  So I think your code for the validator here should look something more like this:
```
assert all(ord(i) < 128 for i in v), 'Name contains non ASCII characters.'
assert all(char not in v for char in (',', ';', '!', '\n', '\t')), \
    'Name contains an invalid character for EnergyPlus ( ,  ;  !  \n  \t).'
assert len(v) > 0, 'Name contains no characters.'
assert len(v) <=100, 'Number of characters in name must be less than 100.'
```

File: app/models/energy/construction.py
Comment (chriswmackey) [inline app/models/energy/construction.py]: We have a typo here. Sorry if this originated from my sample code.

File: app/models/energy/construction.py
Comment (chriswmackey) [inline app/models/energy/construction.py]: This message is not accurate.  The words`valid characters` were used in my original code because we were automatically stripping out the invalid characters.  Now a message like `name contains no characters` or `name is an empty string` is more appropriate.

File: app/models/energy/programtype.py
Comment (chriswmackey) [inline app/models/energy/programtype.py]: Maybe you should do something like this instead of using the word "This":
```
raise ValueError( '"{}" is not a valid entry for latent_fraction'.format(v))
```
That error message will be a little more helpful to people since they will see what their incorrect input was.

File: app/models/energy/simulationparameter.py
Comment (chriswmackey) [inline app/models/energy/simulationparameter.py:13]: For now, I think we should only support the following options since the other ones can be represented with just these five:
```
    timestep = 'Timestep'
    hourly = 'Hourly'
    daily = 'Daily'
    monthly = 'Monthly'
    annual = 'Annual'
```
Also, I would love to run a simulation for a `RumPeriod`. It sounds intoxicating 😄 !

File: app/models/energy/simulationparameter.py
Comment (chriswmackey) [inline app/models/energy/simulationparameter.py]: I know that the EnergyPlus default is 20 but we tend to use 30 in honeybee.  It's really a subjective decision but maybe we should make it 30 for consistency with what we usually do.

File: app/models/energy/simulationparameter.py
Comment (chriswmackey) [inline app/models/energy/simulationparameter.py]: The max is not 31 here.  There's no maximum in the EnergyPlus IDD.

File: app/models/energy/simulationparameter.py
Comment (chriswmackey) [inline app/models/energy/simulationparameter.py]: We start on `Sunday` by default in Honeybee. Not Monday.  Again, I know this is a subjective decision.

File: app/models/energy/simulationparameter.py
Comment (chriswmackey) [inline app/models/energy/simulationparameter.py]: Default should not be 1 as that is too coarse of a timestep to accurately model most phenomena.  The default EnergyPlus timestep is 6 and I think this is a good one.

File: app/models/energy/simulationparameter.py
Comment (chriswmackey) [inline app/models/energy/simulationparameter.py:171]: We should probably have a validator to make sure the timestep is one of these:
https://github.com/ladybug-tools/honeybee-energy/blob/master/honeybee_energy/simulationparameter.py#L32

File: app/models/energy/simulationparameter.py
Comment (chriswmackey) [inline app/models/energy/simulationparameter.py]: I think all of the properties on the `SimulationParameter` schema should be optional in order to make it easy for users to send Models for simulation and just run a typical simulation. Specifically, I think all of the defaults can follow the values in the [simple_simulation_par.json sample](https://github.com/ladybug-tools-in2/energy-model-schema/blob/develop/app/models/samples/json/simple_simulation_par.json)

**Review by chriswmackey — CHANGES_REQUESTED** [review]
Thank you for making a lot of the bigger changes, @tanushree04 .  Please see my comments on both the name validator and the `SimulationParameter` schemas.

File: app/models/energy/programtype.py
Comment (chriswmackey) [inline app/models/energy/programtype.py]: Are we using `re` in this module?  It looks like we are not and we should probably remove it.

File: app/models/energy/schedulebase.py
Comment (chriswmackey) [inline app/models/energy/schedulebase.py]: Again, we should remove the things that we are not using.

File: app/models/energy/scheduleruleset.py
Comment (chriswmackey) [inline app/models/energy/scheduleruleset.py]: One more time.

**Review by chriswmackey — APPROVED** [review]
Thanks, @tanushree04 .  This all looks good and I only had a minor comment about removing imports of dependencies we are not using.  Feel free to merge this once you address them.

File: app/models/energy/simulationparameter.py
Comment (chriswmackey) [inline app/models/energy/simulationparameter.py]: One other small comment.  The default values according to ASHRAE should be 1.25 for heating and 1.15 for cooling.  So I think this one here should be changed to 1.15.

**Comment by tanushree04** [comment]
@chriswmackey The changes to scheduleruleset schema are added in the latest commit as well as the updated example files.

File: app/models/common/datetime.py
Comment (chriswmackey) [inline app/models/common/datetime.py:33]: Why sis you remove this?  This check was useful and the `leap_year` should still be a property of the date.

File: app/models/energy/scheduleruleset.py
Comment (chriswmackey) [inline app/models/energy/scheduleruleset.py:47]: I think we should also change the name of this class to `ScheduleRuleAbridged` if we are going to follow the convention that we have used throughout the schema.

**Review by chriswmackey — APPROVED** [review]
Thanks @tanushree04 .  this looks good except for 2 things:
1. I don't know why you got rid of the `leap_year` property on the date object and the validator on the date object.  It would be great if you could put these back.
2. You should change the name of the `ScheduleRule` class to be `ScheduleRuleAbridged` given the recent change that we made.
Once you do these two things, please merge this in.

File: app/models/common/datetime.py
Comment (tanushree04) [reply app/models/common/datetime.py:33]: @chriswmackey the only place `Date` was being used in the schema was for ScheduleFixedInterval and the `is_leap_year` property is being repeated there https://github.com/ladybug-tools-in2/energy-model-schema/blob/d6c57ecc86b5bf92b2129a945cac32c3ba30fad8/app/models/energy/schedulefixedinterval.py#L46

**Pull request closed without merging**
